client/asset/btc/electrum/network.go

```diff
@@ -0,0 +1,646 @@
+	}
+	peers := make([]*PeersResult, 0, len(resp))
+	for _, peer := range resp {
+		if len(peer) != 3 {
+			fmt.Printf("bad peer data: %v (%T)", peer, peer)
+			continue
+		}
+		addr, ok := peer[0].(string)
+		if !ok {
+			fmt.Printf("bad peer IP data: %v (%T)", peer[0], peer[0])
+			continue
+		}
+		host, ok := peer[1].(string)
+		if !ok {
+			fmt.Printf("bad peer hostname: %v (%T)", peer[1], peer[1])
+			continue
+		}
+		featsI, ok := peer[2].([]interface{})
+		if !ok {
+			fmt.Printf("bad peer feature data: %v (%T)", peer[2], peer[2])
+			continue
+		}
+		feats := make([]string, len(featsI))
+		for i, featI := range featsI {
+			feat, ok := featI.(string)
+			if !ok {
+				fmt.Printf("bad peer feature data: %v (%T)", featI, featI)
+				continue
+			}
+			feats[i] = feat
+		}
+		peers = append(peers, &PeersResult{
+			Addr:  addr,
+			Host:  host,
+			Feats: feats,
+		})
+	}
+	return peers, nil
+}
+
+// SSLPeerAddrs filters the peers slice and returns the addresses in a
+// "host:port" format in separeate slices for SSL-enabled servers and optionally
+// TCP-only hidden services (.onion host names). Note that if requesting to
+// include onion hosts, the SSL slice may include onion hosts that also use SSL.
+func SSLPeerAddrs(peers []*PeersResult, includeOnion bool) (ssl, tcpOnlyOnion []string) {
+peerloop:
+	for _, peer := range peers {
+		isOnion := strings.HasSuffix(peer.Addr, ".onion")
+		if isOnion && !includeOnion {
+			continue
+		}
+		var tcpOnion string // host to accept if no ssl
+		for _, feat := range peer.Feats {
+			if len(feat) < 2 {
+				continue
+			}
+			switch []rune(feat)[0] {
+			case 't':
+				if !isOnion {
+					continue
+				}
+				port := feat[1:]
+				if _, err := strconv.Atoi(port); err != nil {
+					continue
+				}
+				tcpOnion = net.JoinHostPort(peer.Host, port) // hang onto this if there is no ssl
+			case 's':
+				port := feat[1:]
+				if _, err := strconv.Atoi(port); err != nil {
+					continue
+				}
+				addr := net.JoinHostPort(peer.Host, port)
+				ssl = append(ssl, addr) // we know the first rune is one byte
+				continue peerloop
+			}
+		}
+		if tcpOnion != "" {
+			tcpOnlyOnion = append(tcpOnlyOnion, tcpOnion)
+		}
+	}
+	return
+}
+
+// SigScript represents the signature script in a Vin returned by a transaction
+// request.
+type SigScript struct {
+	Asm string `json:"asm"` // this is not the sigScript you're looking for
+	Hex string `json:"hex"`
+}
+
+// Vin represents a transaction input in a requested transaction.
+type Vin struct {
+	TxID      string     `json:"txid"`
+	Vout      uint32     `json:"vout"`
+	SigScript *SigScript `json:"scriptsig"`
+	Witness   []string   `json:"txinwitness,omitempty"`
+	Sequence  uint32     `json:"sequence"`
+	Coinbase  string     `json:"coinbase,omitempty"`
+}
+
+// PkScript represents the pkScript/scriptPubKey of a transaction output
+// returned by a transaction requests.
+type PkScript struct {
+	Asm       string   `json:"asm"`
+	Hex       string   `json:"hex"`
```

Comment:
Ah, true.  Although it would be a very misleading struct definition though.  The `Asm` and `Hex` in this `PkScript` type are the decode pkScript and the hex of the pkScript, neither the `SigScript`, which is something that belongs to an input, not an output.

Here's part of a `blockchain.transaction.get` result:

```json
"vout": [
    {
      "value": 1.21,
      "n": 0,
      "scriptPubKey": {
        "asm": "0 dd8fc00271f7543a7f85960fa94fabd49e451c8d",
        "hex": "0014dd8fc00271f7543a7f85960fa94fabd49e451c8d",
        "reqSigs": 1,
        "type": "witness_v0_keyhash",
        "addresses": [
          "tltc1qmk8uqqn37a2r5lu9jc86jnat6j0y28ydmfuadd"
        ]
      }
    },
```